atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -246,14 +248,21 @@ private CassandraClientPoolImpl(
     }
 
     private void tryInitialize() {
-        cassandra.cacheInitialCassandraHosts();
+        ImmutableMap<CassandraServer, CassandraServerOrigin> initialServers =
+                cassandra.getCurrentServerListFromConfig();
+        ImmutableSet<CassandraServer> cassandraServers = initialServers.keySet();
+
+        cassandra.cacheInitialHostsForCalculatingPoolNumber(cassandraServers);
+
+        setServersInPoolTo(initialServers);
+        Set<CassandraServer> validatedServers = getCachedServers();
+        logStartupValidationResults(cassandraServers, validatedServers);
 
         // for testability, mock/spy are bad at mockability of things called in constructors
         if (startupChecks == StartupChecks.RUN) {
             runOneTimeStartupChecks();
         }
-        refreshPool(); // ensure we've initialized before returning
-        scheduleNextRefresh(0);
+        runAndScheduleNextRefresh(0);
```

Comment:
I preserved this, although we don't propagate the exception if _this_ fails, since we have validated servers from config. The other option was to just use the config servers for 2 minutes / whatever the configured time is, but I figured getting the token range is the better approach